warmrestart/warmRestartHelper.cpp

```diff
@@ -0,0 +1,358 @@
+#include <cassert>
+#include <sstream>
+
+#include "warmRestartHelper.h"
+
+
+using namespace swss;
+
+
+WarmStartHelper::WarmStartHelper(RedisPipeline      *pipeline,
+                                 ProducerStateTable *syncTable,
+                                 const std::string  &syncTableName,
+                                 const std::string  &dockerName,
+                                 const std::string  &appName) :
+    m_restorationTable(pipeline, syncTableName, false),
```

Comment:
What is the reason for using external pipeline for m_restorationTable here? m_restorationTable is for get only.  Asking the caller to provide pipeline brings no benefit.

Reply:
Not sure i got this one. How can we create the table without its pipeline? You mean, using the other table constructor? -- Table(const DBConnector *db, const std::string &tableName). Even if that's what you imply, we would need to create a DBConnector if we don't want the caller to provide it, right?

Please clarify...

Reply:
Sorry, my previous comment is confusing.    WarmStartHelper() has both pipeline and syncTableName as parameters,  the only purpose of them is to prepare a Table handle in WarmStartHelper class.  Could a Table handle be prepared by the callers directly and remove pipeline/syncTableName, just like what has been done for "ProducerStateTable *syncTable"?